Can't resolve '../utils/logger' in 4.6.0

### Environment

 System:
    OS: macOS 12.4
    CPU: (4) x64 Intel(R) Core(TM) i5-7267U CPU @ 3.10GHz
    Memory: 1003.14 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.0.0 - /usr/local/bin/node
    npm: 8.10.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 102.0.5005.115
    Firefox: 101.0.1
    Safari: 15.5
  npmPackages:
    next: 12.1.6 => 12.1.6
    next-auth: 4.6.0 => 4.6.0
    react: 18.2.0 => 18.2.0

### Reproduction URL

none

### Describe the issue

The next compile failes with 
```
error - ./node_modules/next-auth/react/index.js:37:0
Module not found: Can't resolve '../utils/logger'
```
after updating to 4.6.0, reverting to 4.5.0 fixes the issue

### How to reproduce

- Update next-auth to 4.6.0 from 4.5.0
- The next build fails

### Expected behavior

- Build should not fail

--- a/packages/next-auth/config/babel.config.js
+++ b/packages/next-auth/config/babel.config.js
@@ -25,7 +25,6 @@ module.exports = (api) => {
     ignore: [
       "../src/**/__tests__/**",
       "../src/adapters.ts",
-      "../src/core/types.ts",
       "../src/providers/oauth-types.ts",
     ],
     comments: false,

--- a/packages/next-auth/package.json
+++ b/packages/next-auth/package.json
@@ -62,7 +62,8 @@
     "index.js",
     "adapters.d.ts",
     "middleware.d.ts",
-    "middleware.js"
+    "middleware.js",
+    "utils"
   ],
   "license": "ISC",
   "dependencies": {

--- a/packages/next-auth/src/core/types.ts
+++ b/packages/next-auth/src/core/types.ts
@@ -14,7 +14,7 @@ import type { CookieSerializeOptions } from "cookie"
 
 import type { NextApiRequest, NextApiResponse } from "next"
 
-import { InternalUrl } from "../utils/parse-url"
+import type { InternalUrl } from "../utils/parse-url"
 
 export type Awaitable<T> = T | PromiseLike<T>
 

--- a/packages/next-auth/tsconfig.json
+++ b/packages/next-auth/tsconfig.json
@@ -1,7 +1,7 @@
 {
   "extends": "@next-auth/tsconfig/base.json",
   "compilerOptions": {
-    // "emitDeclarationOnly": true,
+    "emitDeclarationOnly": true,
     "strictNullChecks": true,
     "lib": ["dom", "dom.iterable", "esnext"],
     "allowJs": true,

--- a/packages/next-auth/tsconfig.json
+++ b/packages/next-auth/tsconfig.json
@@ -1,7 +1,7 @@
 {
   "extends": "@next-auth/tsconfig/base.json",
   "compilerOptions": {
-    "emitDeclarationOnly": true,
+    // "emitDeclarationOnly": true,
     "strictNullChecks": true,
     "lib": ["dom", "dom.iterable", "esnext"],
     "allowJs": true,
@@ -14,7 +14,8 @@
     "stripInternal": true,
     "skipDefaultLibCheck": true,
     "baseUrl": ".",
-    "outDir": "."
+    "outDir": ".",
+    
   },
   "exclude": ["./*.js", "./*.d.ts", "config", "**/__tests__", "tests"]
 }

